Tasks/DotNetCoreCLIV2/dotnetcore.ts

```diff
@@ -318,29 +319,62 @@ export class dotNetExe {
     }
 
     private getProjectFiles(): string[] {
-        var projectPattern = this.projects;
-        var searchWebProjects = this.isPublishCommand() && this.publishWebProjects;
+        let projectPattern = this.projects;
+        const searchWebProjects: boolean = this.isPublishCommand() && this.publishWebProjects;
         if (searchWebProjects) {
-            projectPattern = ["**/*.csproj", "**/*.vbproj", "**/*.fsproj"];
+            projectPattern = ['**/*.csproj', '**/*.vbproj', '**/*.fsproj'];
         }
 
-        var projectFiles = utility.getProjectFiles(projectPattern);
+        const projectFiles = utility.getProjectFiles(projectPattern);
+        let resolvedProjectFiles: string[] = [];
 
         if (searchWebProjects) {
-            projectFiles = projectFiles.filter(function (file, index, files): boolean {
-                var directory = path.dirname(file);
-                return tl.exist(path.join(directory, "web.config"))
-                    || tl.exist(path.join(directory, "wwwroot"));
+            resolvedProjectFiles = projectFiles.filter(function (file, index, files): boolean {
+                const directory = path.dirname(file);
+                return tl.exist(path.join(directory, 'web.config'))
+                    || tl.exist(path.join(directory, 'wwwroot'));
             });
 
-            if (!projectFiles.length) {
-                tl.error(tl.loc("noWebProjctFound"));
+            if (!resolvedProjectFiles.length) {
+                projectFiles.forEach((projectFile) => {
```

Comment:
This is wrong. You should use filter to get array. Right now you are just assigning the last element to match to resolvedProjectFilers

Reply:
Add UT for this